### PR TITLE
Checkpoint missing tables

### PR DESCRIPTION
**Pull request by AmishgaAlphonius**
# Description of the problem

- A few tables were omitted from being _"checkpointed"_ causing a lost of valuable information when simulations were restarted (Issue: #916).

# Description of the solution

- The missing tables have been added to the `write_checkpoint()` and `read_checkpoint()` of each physic.

# How Has This Been Tested?

- CI
- Restart application-tests were updated:
  - [x] `applications_tests/lethe-fluid/cavity_ht_average_velocity_restart`
  - [x] `applications_tests/lethe-fluid/mms-transient-conduction-restart_gls`
  - [x] `applications_tests/lethe-fluid/taylor-green-vortex-restart_gls_bdf1`

# Documentation

- A small correction was made to `doc/source/parameters/cfd/restart.rst`

### CODE REVIEW

**Review by blaisb — CHANGES_REQUESTED** [review]
A quick question. I'll try finishing the review of the rest ASAP

File: include/core/utilities.h [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #define lethe_utilities_h
 
 #include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/convergence_table.h>
```
Comment (blaisb) [inline include/core/utilities.h:24]:
```suggestion
```

**Review by blaisb — APPROVED** [review]
Just one line to remove and it should be good :)

File: source/solvers/heat_transfer.cc [hunk 2]
```diff
@@ -940,6 +940,33 @@ HeatTransfer<dim>::write_checkpoint()
...
+    }
+  if (this->simulation_parameters.post_processing
+        .calculate_temperature_statistics)
+    {
+      serialize_table(
+        this->statistics_table,
+        prefix +
+          this->simulation_parameters.post_processing.temperature_output_name +
+          suffix);
+    }
```
Comment (oguevremont) [inline source/solvers/heat_transfer.cc:969]: Is it intended that there are not `{}` for the first if, but that there are for the other two?

File: source/solvers/heat_transfer.cc [hunk 3]
```diff
@@ -972,6 +999,33 @@ HeatTransfer<dim>::read_checkpoint()
...
+  if (this->simulation_parameters.post_processing
+        .calculate_temperature_statistics)
+    {
+      deserialize_table(
+        this->statistics_table,
+        prefix +
+          this->simulation_parameters.post_processing.temperature_output_name +
+          suffix);
+    }
```
Comment (oguevremont) [inline source/solvers/heat_transfer.cc:1028]: Same question here

**Review by oguevremont — APPROVED** [review]

File: source/solvers/heat_transfer.cc (hunk 3 above)
Comment (oguevremont) [reply source/solvers/heat_transfer.cc:1028]: I saw this a few times and it was always consistent, so not a problem unless you prefer to change it

**Pull request merged**